lib/whitehall_importer/integrity_check_error.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class IntegrityCheckError < AbortImportError
+    attr_reader :problems, :payload
+
+    def initialize(state, problems, payload)
```

Comment:
As per previous comment we can probably handle setting various variables in here based on an integrity_check itself rather than individual attributes

Reply:
Now passing the `IntegrityChecker` instance instead of separate problems and payload.